spartan/src/polycommitments/mod.rs

```diff
@@ -41,13 +41,23 @@ pub trait PolyCommitmentTrait<G: CurveGroup>:
   + AddAssign<Self>
   + MulAssign<G::ScalarField>
   + Mul<G::ScalarField, Output = Self>
+  + Into<Vec<G>>
+  + From<Vec<G>>
   + Default
   + Clone
   + Send
   + Sync
 {
   // this should be the commitment to the zero vector of length n
   fn zero(n: usize) -> Self;
+
+  fn into_affine(self) -> Vec<G::Affine> {
+    self.into().iter().map(|c| c.into_affine()).collect()
+  }
+
+  // for commitments in the form of a single field element (e.g., Zeromorph) this should return it
+  // for commitments unable to be representated as such (e.g., Hyrax) it should panic unimplemented!()
+  fn into_single(self) -> G;
```

Comment:
Wouldn't having just one method
```rust
/// Convert commitment into an affine point.
///
/// This method should only be implemented in case the commitment scheme returns a single curve point.
fn into_affine(self) -> Option<G::Affine> {
    None
}
```
be sufficient?

Looks cleaner to me. We can then panic inside the folding scheme implementation. Returning error doesn't make sense as it cannot be handled as such -- the commitment scheme is not suitable.